### PR TITLE
[feat]: [CI-12836]: Add dlc metadata to savings response

### PR DESCRIPTION
**Pull request by anurag-harness**

### CODE REVIEW

File: types/savings.go [hunk 1]
```diff
@@ -36,5 +36,6 @@ type SavingsOverview struct {
 
 // This Structure will have the savings overview for each step and also detailed metrics in the future such as dlc metrics and gradle metrics
 type SavingsResponse struct {
-	Overview []SavingsOverview `json:"overview"` // array of SavingsOverview since one step can have multiple savings features enabled
+	Overview   []SavingsOverview `json:"overview"` // array of SavingsOverview since one step can have multiple savings features enabled
+	DlcMetrics dlc.Metrics       `json:"dlc_metrics"`
```
Comment (rutvijmehta-harness) [inline types/savings.go:40]: Why is this required?

**Review by rutvijmehta-harness — CHANGES_REQUESTED** [review]

File: types/savings.go (hunk 1 above)
Comment (anurag-harness) [reply types/savings.go:40]: To return dlc metrics to the UI: https://app.harness.io/ng/account/vpCkHKsDSxK9_KYfjCTMKA/module/code/orgs/default/projects/CIPRChecks/repos/harness-ti/pulls/225

Comment (anurag-harness) [reply types/savings.go]: Added a new dlc metadata struct which I will use in SavingsResponse instead

**Review by rutvijmehta-harness — APPROVED** [review]
